src/main/kotlin/com/openlattice/mechanic/integrity/CheckOrphanedEntitySets.kt

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2019. OpenLattice, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+package com.openlattice.mechanic.integrity
+
+import com.google.common.base.Stopwatch
+import com.openlattice.authorization.securable.SecurableObjectType
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.postgres.PostgresArrays
+import com.openlattice.postgres.PostgresColumn.*
+import com.openlattice.postgres.PostgresTable
+import com.openlattice.postgres.PostgresTable.*
+import com.openlattice.postgres.ResultSetAdapters
+import com.openlattice.postgres.streams.BasePostgresIterable
+import com.openlattice.postgres.streams.PreparedStatementHolderSupplier
+import com.openlattice.postgres.streams.StatementHolderSupplier
+import org.slf4j.LoggerFactory
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+class CheckOrphanedEntitySets(private val toolbox: Toolbox) : Check {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(CheckOrphanedEntitySets::class.java)
+    }
+
+    override fun check(): Boolean {
+        logger.info("Starting to delete entries of orphaned entity sets from securable object types, permissions, " +
+                "materialized entity sets and entity set property metadata tables.")
+        val sw = Stopwatch.createStarted()
+
+
+        // check securable objects and permissions
+
+        var deletedEntitySetIdsInSecurableObjects = setOf<UUID>()
+
+        val deletedPermissionCount = toolbox.hds.connection.use { conn ->
+            conn.prepareStatement(deleteOrphanedPermissions).use { ps ->
```

Comment:
might be faster and less work to chain this using CTEs in the database:

with deletedEntitySetIds as ($deleteOrphanedSecurableObjects)
deleteOrphanedPermissions where acl_key in (select acl_key from deletedEntitySetIds)

If you want specific counts you can do

with deletedSecureableObjects as ($deleteOrphanedSecurableObjects),
with deletedPermissions as ($deleteOrphanedPermissions where acl_key in (select acl_key from deletedSecureableObjects))
select * from (select count(*) as securableObjects from deletedSecureableObjects) a, (select count(*) as permissions from deletedPermissions) b